.github/workflows/docker.yml

```diff
@@ -5,45 +5,133 @@ on:
 
 env:
   CI_RUST_TOOLCHAIN: 1.70.0
+  REGISTRY_IMAGE: datenlord/xline
 
 jobs:
-  docker:
-    name: Docker
-    runs-on: ubuntu-latest
+  push_image:
+    name: Push Docker Image
+    runs-on: ${{ matrix.job.os }}
+    outputs:
+      app_version: ${{ steps.generate_app_version.outputs.app_version }}
+    strategy:
+      fail-fast: false
+      matrix:
+        job:
+          - os: ubuntu-latest
+            platform: linux/amd64
+            target: x86_64-unknown-linux-gnu
+            cross_image: x86_64-linux-gnu
+          - os: ubuntu-latest
+            platform: linux/arm64
+            target: aarch64-unknown-linux-gnu
+            cross_image: aarch64-linux-gnu
     steps:
       - name: Checkout
-        uses: actions/checkout@v2
-      - run: sudo bash ./.github/scripts/install_deps.sh
-      - uses: actions-rs/toolchain@v1
+        uses: actions/checkout@v3
+
+      - name: Generate App Version
+        id: generate_app_version
+        run: echo app_version=`git describe --tags --always` >> $GITHUB_OUTPUT
+
+      - name: Set up Docker Buildx
+        uses: docker/setup-buildx-action@v1
+
+      - name: Setup Custom Image for ${{ matrix.job.cross_image }}
+        uses: docker/build-push-action@v2
+        with:
+          context: .github/scripts
+          file: .github/scripts/Dockerfile.${{ matrix.job.target }}
+          tags: ${{ matrix.job.cross_image }}:latest
+          load: true
+          cache-from: type=gha
+          cache-to: type=gha,mode=max
+
+      - name: Install Toolchain
+        uses: dtolnay/rust-toolchain@master
```

Comment:
I think we should consider using stable version instead of master

Reply:
done